### PR TITLE
compute - bump network_performance_config to ga

### PR DESCRIPTION
**Pull request by DrFaust92**
If this PR is for Terraform, I acknowledge that I have:

- [x] Searched through the [issue tracker](https://github.com/hashicorp/terraform-provider-google/issues) for an open issue that this either resolves or contributes to, commented on it to claim it, and written "fixes {url}" or "part of {url}" in this PR description. If there were no relevant open issues, I opened one and commented that I would like to work on it (not necessary for very small changes).
- [x] Ensured that all new fields I added that can be set by a user appear in at least one [example](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/templates/terraform/examples) (for generated resources) or [third_party test](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/third_party/terraform/tests) (for handwritten resources or update tests).
- [x] [Generated Terraform providers](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/generate-providers.md), and ran [`make test` and `make lint`](https://googlecloudplatform.github.io/magic-modules/docs/getting-started/run-provider-tests/#run-unit-tests) in the generated providers to ensure it passes unit and linter tests.
- [x] [Ran](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/docs/content/docs/getting-started/run-provider-tests.md) relevant acceptance tests using my own Google Cloud project and credentials (If the acceptance tests do not yet pass or you are unable to run them, please let your reviewer know).
- [x] Read the [Release Notes Guide](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/.ci/RELEASE_NOTES_GUIDE.md) before writing my release note below.

**Release Note Template for Downstream PRs (will be copied)**

```release-note:enhancement
compute: - promoted field `network_performance_config` to ga 
```

### CODE REVIEW

**Review by hao-nan-li — COMMENTED** [review]
Thanks for `promoting network_performance_config` to ga.

Could you point me to a doc, or anything that shows this field is promoted to GA by the service team?

**Comment by DrFaust92** [comment]
Hi @hao-nan-li 

compute engine v1 docs https://cloud.google.com/compute/docs/reference/rest/v1/instances/insert

File: mmv1/third_party/terraform/resources/resource_compute_instance.go.erb [hunk 1]
```diff
@@ -1260,11 +1258,9 @@ func resourceComputeInstanceRead(d *schema.ResourceData, meta interface{}) error
 	if err := d.Set("machine_type", tpgresource.GetResourceNameFromSelfLink(instance.MachineType)); err != nil {
 		return fmt.Errorf("Error setting machine_type: %s", err)
 	}
-<% unless version == 'ga' -%>
```
Comment (hao-nan-li) [inline mmv1/third_party/terraform/resources/resource_compute_instance.go.erb:1263]: I searched in the file and it looks like there are couple more lines to be removed, could you search `networkPerformanceConfig` in this file to clean them up?

Line 1069, 1095

File: mmv1/third_party/terraform/resources/resource_compute_instance_template.go.erb [hunk 2]
```diff
@@ -1596,11 +1594,9 @@ func resourceComputeInstanceTemplateRead(d *schema.ResourceData, meta interface{
 	if err = d.Set("project", project); err != nil {
 		return fmt.Errorf("Error setting project: %s", err)
 	}
-<% unless version == 'ga' -%>
```
Comment (hao-nan-li) [inline mmv1/third_party/terraform/resources/resource_compute_instance_template.go.erb:1599]: Same for this file. 

Line 1206, 1227

File: mmv1/third_party/terraform/resources/resource_compute_region_instance_template.go.erb [hunk 3]
```diff
@@ -1127,11 +1125,9 @@ func resourceComputeRegionInstanceTemplateRead(d *schema.ResourceData, meta inte
 	if err = d.Set("project", project); err != nil {
 		return fmt.Errorf("Error setting project: %s", err)
 	}
```
Comment (hao-nan-li) [inline mmv1/third_party/terraform/resources/resource_compute_region_instance_template.go.erb:1127]: Same for this file

line 926, 946

**Review by hao-nan-li — COMMENTED** [review]
Thanks for pointing me to the right doc, left couple commands in the files as it looks to have couple more lines to remove.

**Comment by DrFaust92** [comment]
[hao-nan-li](https://github.com/hao-nan-li) good catch, fixed

**Review by hao-nan-li — APPROVED** [review]
LGTM

**Pull request merged**